### ISSUE TITLE
Render place names on areas the same way as on nodes

### ISSUE THREAD

**Issue by joakimfors** [gravitystorm/openstreetmap-carto#103]
Currently place names are only "correctly" rendered when mapped on nodes. Place names should be rendered in the same way when placed on closed ways and relations.

**Comment by xenos1984** [comment]
Is there any progress on this issue?

Just another thought I would like to add, although I don't know whether this would be working: Many place=* areas are mapped as multipolygons with an additional node with role admin_centre. Would it be possible to check for such a member and render the place name for the area only once at this point?

**Comment by math1985** [comment]
> Is there any progress on this issue?

No, there is still no agreement on whether this is a good idea or not.

**Comment by skylerbunny** [comment]
Ran into this, while discovering that for place=(suburb|village|hamlet|neighbourhood|locality|isolated_dwelling|farm), points render and areas do not, so now I'm stuck trying to figure out how to make Mapnik render the following non-administrative census designated place, which has a defined area, as a named place:

http://www.openstreetmap.org/way/389991209

This can also be found in neighborhoods with defined edges. Examples: ways 388843930 and 313909239.

My take on this:
1. This issue is not talking about administrative areas, nor should it be confused for doing so. That is completely covered in https://github.com/gravitystorm/openstreetmap-carto/issues/104 , discussing how and where to place a label involving administrative boundary relations and its members.
2. Numerous cases exist where a non-administrative boundary that is ALSO one of the set of (suburb|village|hamlet|neighbourhood|locality|isolated_dwelling|farm), is a documentable area rather than a point: see above.
3. All of the above show on the Wiki as having acceptable use on areas. If it's 'not a good idea to render areas' with place=(the above), then further pollution needs to be stopped by declaring areas as invalid structures to hold those place types. What guidance do people receive to not use areas, other than that Mapnik won't show their names?

If and until (3) happens, I can't really see the downside in displaying the names of these areas at their geographical center(s) (area or multipolygon). Being able to identify multipolygons with nodes as members for labelling might be nice, but an appropriate first step IMHO would be rendering a name at the center of an area of these place names. Those place names not being in an ideal spot isn't ideal, but at least prevents people double tagging place nodes for the renderer, which I don't want to do either.

Failing that, what advice can be given for what I should do tagging these cases?

**Comment by pnorman** [comment]
> Failing that, what advice can be given for what I should do tagging these cases?

For tagging, you should ask help.openstreetmap.org, the tagging list, or somewhere else. If we change what we render it won't change what the appropriate tagging is.

**Comment by skylerbunny** [comment]
Understood on that point. If the fallback goes to 'It's not us' (entirely fair), the standing consensus (both prescriptively and 'in the field') already is that the appropriate tagging for these place types is both nodes and areas.

**Comment by dieterdreist** [comment]
sent from a phone

> Am 07.01.2016 um 06:35 schrieb skylerbunny <notifications@github.com>:
> 
> which has a defined area, as a named place:
> 
> http://www.openstreetmap.org/way/389991209
> 


while I advocate place areas, this one looks odd. Why is this strange part of the wetland part of the settlement? IMHO a place area should cover the settlement but not parts that clearly don't belong to  it.

**Comment by skylerbunny** [comment]
> Why is this strange part of the wetland part of the settlement?

Aha. Basically, 'because the U.S. census bureau said so'. It's a census designated place. I added the census.gov source PDF (and a boundary=census) tag to the way, have a look.

That being fixed/sourced, of course, I don't want the topic to stray too much into why place areas are what they are, and want to stay zeroed in on the fact that Mapnik doesn't render their names.

**Comment by dieterdreist** [comment]
2016-01-07 18:31 GMT+01:00 skylerbunny <notifications@github.com>:

> Aha. Basically, 'because the U.S. census bureau said so'. It's a census
> designated place. I added the census.gov source PDF (and a
> boundary=census) tag to the way, have a look.
>
> That being fixed/sourced, of course, I don't want the topic to stray too
> much into why place areas are what they are, and want to stay zeroed in on
> the fact that Mapnik doesn't render their names.
>

For discussions what place areas are, we should likely move over to the
tagging ML. From my point of view, boundaries from official authorities
which do not describe settlements, but are basically orthogonal
administrative boundaries (e.g. used for statistical purposes) should get a
different tag and not be mapped as place areas. From a socio-geographic
viewpoint, these wetlands and farmlands are not part of the hamlet (or more
general settlement), as they are neither built up, nor encircled or in a
similar way delimited.

**Comment by skylerbunny** [comment]
> For discussions what place areas are, we should likely move over to the tagging ML.

**Exactly.** The *death* of this bug report (as it has been in the past, above) has been straying into discussions appropriate for the tagging ML.

The tl;dr of issue 103 has been 'regardless of where I want to tag place names, and even whether correct or not - *all* of which is tagging ML discussion anyway - their names don't render in the Mapnik scheme'.

**Comment by dieterdreist** [comment]
2016-01-07 18:53 GMT+01:00 skylerbunny <notifications@github.com>:

> The tl;dr of issue 103 has been 'regardless of where I want to tag place
> names, and even whether correct or not - *all* of which is tagging ML
> discussion anyway - their names don't render in the Mapnik scheme'.
>


yes, but if you post inappropriately tagged objects as examples it is hard
to discuss appropriate rendering solutions.

**Comment by skylerbunny** [comment]
I was thinking on this: How about this, as a 'reasonable first pass'?

1. place=(suburb|village|hamlet|neighbourhood|locality|isolated_dwelling|farm) gets a text label in the center of each distinct area it encompasses, at the same text size a node of the same type would already get. For simple areas, this means one label; for multipolygons, it means multiple text labels, one for each area. I refer here, to the same method of name labeling that is used for leisure=nature reserve (see http://www.openstreetmap.org/relation/5012497 ).
2. We do NOT address/change in any way border lines or area coloration *of* these areas in this issue, and doing (1) means this issue can be marked closed.

My guess, anyway, is that exposing the very existence of these areas to Mapnik users/editors by doing (1) will prompt discussion about just *how* to display their edges - and if appropriate, generate discussion on the ML about how the areas are being tagged.

**Comment by dieterdreist** [comment]
2016-01-07 21:00 GMT+01:00 skylerbunny <notifications@github.com>:

> I was thinking on this: How about this, as a 'reasonable first pass'?
>
>    1. place=(suburb|village|hamlet|neighbourhood|locality|isolated_dwelling|farm)
>    gets a text label in the center of each distinct area it encompasses, at
>    the same text size a node of the same type would already get.
>
>

what about town and city? "locality" is a placeholder for more specific
placenames, but not to be used on settlements/inhabited places, so I'd
remove it from this list.



>
>    1.
>
> My guess, anyway, is that exposing the very existence of these areas to
> Mapnik users/editors by doing (1) will prompt discussion about just *how*
> to display their edges - and if appropriate, generate discussion on the ML
> about how the areas are being tagged.


rather than the edges I'd use the polygons for rendering built-up areas in
low to low-medium zoom levels instead of (landuses and) the shapefile.

**Comment by gravitystorm** [comment]
> the fact that Mapnik doesn't render their names

Just as an aside, please refer to these stylesheets as "openstreetmap-carto", not "Mapnik" - Mapnik is a rendering library used by many different stylesheets, and mixing names around confuses and likely discourages people from contributing.

**Comment by skylerbunny** [comment]
> Just as an aside, please refer to these stylesheets as "openstreetmap-carto"

No problem, my mistake. Sorry about that.

I took that list above from the source, from https://github.com/gravitystorm/openstreetmap-carto/blob/6ac399d63876cfe03f25f324b4023e7a329d4ee2/project.yaml#L1537

Looking right above that line, I can see that 'city' and 'town' would in fact also be included in this list if their areas don't already have names rendered elsewhere in openstreetmap-carto's source code.

Re locality areas, http://wiki.openstreetmap.org/wiki/Tag:place%3Dlocality indicates that an area is a valid use, and it is used in fairly frequently in Europe, SE and Western United States, and Alaska (7100 cases). http://overpass-turbo.eu/s/dD5 . Having said that, the wiki DOES only talks about localities used on nodes, and I agree in principle with your point. It could be argued that a fix for this is to simply edit the wiki to disallow locality on areas in the infobox to make that page consistent.

**Comment by pnorman** [comment]
Because we've changed how we render place names since this issue was raised, it now depends on the schema change for capital and population for polygons.

**Comment by Hjart** [comment]
place=* area names are still not rendered (see i.e. (http://www.openstreetmap.org/way/492177524). Personally I much prefer to have them actually be areas rather than just nodes, so I'll really really appreciate finally having them rendered. I don't understand how this can still be an issue.

**Comment by dieterdreist** [comment]
2017-06-10 19:11 GMT+02:00 Michael Andersen <notifications@github.com>:

> place=* area names are still not rendered (see i.e. (
> http://www.openstreetmap.org/way/492177524). Personally I much prefer to
> have them actually be areas rather than just nodes, so I'll really really
> appreciate finally having them rendered. I don't understand how this can
> still be an issue.
>

My guess is some people do not want mappers to map places (only) as areas,
and current usage is not consistent for all values, e.g.
1. could lead to edit wars sometimes
2. mappers might omit the central node (i.e. loss of information)
3. sometimes place tags are added to administrative entities, where the
place arguably only covers part of the administrative entity
4. difficulty to avoid duplicate names if a node and an area are mapped for
the same place

FWIW, I'm in favor of rendering names from place polygons, but duplicate
names should be avoided if possible (even if there's a node and an area).

**Comment by jojo4u** [comment]
I have drafted an article with some scenarios: https://wiki.openstreetmap.org/wiki/Proposed_features/place_areas
My preferred solution would be scenario A: The introduction of boundary=place on closed ways (areas) and of tag boundary=place for type=boundary relations. The place nodes are put in the relation as role place_centre.

**Comment by Pikse** [comment]
This bug should be about simply making it possible to show place name on centroid when place=* is set in combination with whatever keys deemed appropriate and in regions where appropriate.

Obviously settlements are understood differently in different regions, as shown above. Some countries have settlements as officially designated territorial units (not administrative units), other don't. Some regions are considered to have [dispersed settlements](https://en.wikipedia.org/wiki/Dispersed_settlement), in others separate household are not considered to be part of any settlement. Map should simply reflect that, so this can't be an issue here.

**Comment by stevenLAD** [comment]
I am having difficulty understanding this issue. There appears to be some consensus that areas should be named as it renders the maps more usable (and meaningful IMHO). I have proposed a solution but that was rejected as closed (2792) - at present, as someone has already said in this thread, the renderer ignores polygons when looking for place names.

The issue appears to be duplicity between areas and nodes, but as long as this is not resolved, that duplicity will only increase as people try to workaround the problem by adding nodes to areas, and by using landuse=residential (which works but is not the OSM approved way).

**Comment by drkludge** [comment]
In #2792 you mentioned placenames.  I recently changed placename tags to name tags.  As far as Keepright said, placename is deprecated use name tag.  But I am not sure that I have answered your question.

**Comment by skylerbunny** [comment]
Original bug, 4 years ago:

> Currently place names are only "correctly" rendered when mapped on nodes. Place names should be rendered in the same way when placed on closed ways and relations.

Today:
> The issue appears to be duplicity between areas and nodes, but as long as this is not resolved, that duplicity will only increase as people try to workaround the problem...

Please consider this...I admit I'm frustrated in advance. 

If this bug had been resolved four years ago, by this time, node and area duplication would have been eliminated years ago by way of iterative cleanup. The only reason this bug still is open IMHO is because there is a faction that continues to argue whether areas should be mapped, when the community  decided de-facto years ago that they would be. The wiki mentions it as a supported use case. It's so widespread that a special note exists on the place= wiki page referencing this bug to explain why it doesn't render. What exactly is the problem with implementing this?

No amount of principled or wishful thinking will prevent people from using place tags on areas, or address issues with doing so through discussion _here, in this bug_. If the issue is duplication, expose the issue to the mapping community by rendering the areas and nodes both.  They're  there whether we like it or not.

Users WILL resolve duplication over time, or find better,  more specific non place area tags over time. Maybe new tags will even be created to address edge cases.  But users need to be made aware they're even there. Thus, rendering.

**Comment by kocio-pl** [comment]
I feel that it should be rendered, because place is basically a name - so it relates to any geometric entity (for example #2673). See also https://en.wikipedia.org/wiki/Toponymy .

**Comment by AragonChristopherR17Z** [comment]
What I do is make a multipolygon of, for example, a neighborhood and make a relation to a named neighborhood point as the label. :P
For example: https://www.openstreetmap.org/relation/7543322

**Comment by skylerbunny** [comment]
@kocio-pl I also replied related to this at https://wiki.openstreetmap.org/wiki/Talk:Proposed_features/place_areas , whose proposal seems to try to solve about three problems at once. I suggest we solve one problem - render the areas, the one in this bug. It will lend visibility to those other problems (like 'do we even want to be using tag X?' or 'but we don't have a way that openstreetmap-carto renders a label for an area', and 'but this will cause name rendering duplication') on its own. Any of those can be solved through, and a solution for them will almost certainly be accelerated by -- rendering the areas.

In the above wiki location, I suggested using the same method of rendering as we do for land_use multipolygons, probably without a special border/fill color. It's easy, it gets the job done, and it will move this forward.

**Comment by chrisfleming** [comment]
We do really need to get this sorted, I did propose some kind of improvemnt 3 years ago: https://github.com/gravitystorm/openstreetmap-carto/pull/427 - It got pulled into a bigger update that never made it through. 

Lots of data users DO render placename based on node and or area and in those cases names are rendered twice. Where I am (Edinburgh) the plan was to 1. add area's across the city, 2. Get rendering working on areas before 3. removing the "center" nodes. We have never got past step 2. 

Furthermore it seems to me that the default mapbox themes, cycle.travel, and the cycle map all do render both the area name and the node names.

**Comment by lukaszkruk** [comment]
Joining the many who cry for this to be fixed. 

I'm suggesting the simplest possible solution: name of a place drawn as a closed way should be styled like name of a place drawn as a node. This is the scope of the original bug report.

Are there any practical steps that should be taken to achieve this? Is there a need for a vote or some other procedure?

**Comment by pnorman** [comment]
> Are there any practical steps that should be taken to achieve this? Is there a need for a vote or some other procedure?

No, a pull request is needed.

**Comment by mboeringa** [comment]
See my remark on the pull:

https://github.com/gravitystorm/openstreetmap-carto/pull/2816#issuecomment-328298417

**Comment by Komzpa** [comment]
I believe the only way this could be done, compatible with both node-tagging, area-tagging and mixed-tagging, is like this:

https://github.com/gravitystorm/openstreetmap-carto/pull/2939

If anyone wants names on polygons rendered please get this PR un-closed by maintainers and merged.

**Comment by vbertola** [comment]
I just want to add a couple of points continuing the thread on #2816.

@imagico still fails to understand that there are places which have defined borders and yet are not administrative units or landuses or other existing types, but just... places. I can bring my example once again: the suburbs in Turin (a 1-million-people city) were administrative units until 1985, but they were then grouped into bigger boroughs (that of course we mapped as administrative units). We cannot map the suburbs as administrative units, as they are not that any more. However, the suburbs still have well known borders from that time, and are still in use, much more than the boroughs, as place names. They are still used as the basis for civic committees, working groups and so on, so knowing exactly whether a street is in this or that suburb is useful and makes a difference. How in the hell could I map these places precisely as a node? Again, I'm fine if I am required to add a node as admin_centre as a tagging fiction, even if there is no "admin" and no "centre" in the suburb, but please stop saying that all people that want to map a place as an area are just wrong and their areas are fake and "meaningless". Maybe this is true in your surroundings, but it is not true everywhere.

Also, how reasonable it is to continuously challenge the fact that there is "inconsistency" in places-as-areas because some people draw areas around built-up land while others around the entire territory or around some other geometry, while at the same time there is the same inconsistency in places-as-nodes, with some people putting the node at the centroid of the area while others put it in the main square and others on the town hall and so on, but this is not considered a problem? I really don't understand.

**Comment by dieterdreist** [comment]
Am Di., 8. Jan. 2019 um 10:12 Uhr schrieb Vittorio Bertola <
notifications@github.com>:

> I just want to add a couple of points continuing the thread on #2816
> <https://github.com/gravitystorm/openstreetmap-carto/pull/2816>.
>
> @imagico <https://github.com/imagico> still fails to understand that
> there are places which have defined borders and yet are not administrative
> units or landuses or other existing types, but just... places. I can bring
> my example once again: the suburbs in Turin (a 1-million-people city) were
> administrative units until 1985, but they were then grouped into bigger
> boroughs (that of course we mapped as administrative units).
>


in Rome there are also different layers of named areas, which only
sometimes coincide with administrative subdivisions.

And you can often define neighbourhoods outside the centre core by looking
at the structure: areas developped in the same time, often as one project
(same lead architect/s, same style, etc.). Also in centres you might be
able to identify specific quarters or neighbourhoods with specific
properties, e.g. from medieval professions like tanners or dyers, jewish
ghettoes, etc., or more recently settlement parts for factory workers,
entertainment districts, ...

**Comment by Pikse** [comment]
> @dieterdreist And you can often define neighbourhoods outside the centre core by looking at the structure: areas developped in the same time, often as one project (same lead architect/s, same style, etc.). Also in centres you might be able to identify specific quarters or neighbourhoods with specific properties ...

These arbitrary or unverifiable areas that may be inappropriately tagged as "place" don't really stand a comparison with non-administrative neighbourhoods that have designated or otherwise well-defined borders. Several other types of places that are valid use cases of place as an area have been mentioned previously here and in comments to related pull requests. Yesterday, I tried to provide a brief overview [here](https://github.com/gravitystorm/openstreetmap-carto/pull/2816#issuecomment-451968280). Some people seem to think that if there are no places (non-administrative, not landuse equivalent) with well-defined borders in region where they live then there are no such places in rest of the world either. Once we get over this misconception, then we'll probably be much closer to solution.

**Comment by jeisenbe** [comment]
Since both PR #2816 (render labels on settlement `place` nodes and areas) and PR #2939 (only render settlement `place` nodes on areas if there is no node with the same tags inside) were rejected, I'm closing this issue. However, if there is another solution that I've overlooked, please feel free to continue discussion or submit a PR.

**Comment by Hjart** [comment]
As a mapper and a data user, I have to say that I'm seriously disappointed that this is getting nowhere ..

**Comment by dawansv** [comment]
> As a mapper and a data user, I have to say that I'm seriously disappointed that this is getting nowhere ..

I too am disappointed this was closed with no guidance. My organization has local teams working in villages in Mali, and over the years we have added vilage names as we collect them in the field. It is quite important to us to be able to litterally put these villages on the map by naming them and a great complement to the incredible work done by the mapping community using satelitte imagery. So our contribution is very limited but important I think. What kind of guidance is given to people like us? Now we have a situation were some more experience mappers are merging node information with areas. The rendering is completely different. But we are certainly in no position to give these more experienced mappers any lesson or guidance. 

Look at that area south of Bamako for instance. 
https://www.openstreetmap.org/#map=9/12.3024/-7.6382
Villages that have names as nodes appear around zoom level 9. But a lot of villages that have been named are no longer labelled until your reach zoom level 13 because their name nodes have been merged into their areas. 

What are we supposed to do with that?  Are we supposed to manually undo the work of more experience mappers and re-add duplicate points to areas that don't have them? That doesn't make any sense since there is no guidance telling mappers NOT to put names into areas.

**Comment by lukaszkruk** [comment]
> > As a mapper and a data user, I have to say that I'm seriously disappointed that this is getting nowhere ..
> 
> I too am disappointed this was closed with no guidance. My organization has local teams working in villages in Mali, and over the years we have added vilage names as we collect them in the field. It is quite important to us to be able to litterally put these villages on the map by naming them and a great complement to the incredible work done by the mapping community using satelitte imagery. So our contribution is very limited but important I think. What kind of guidance is given to people like us? Now we have a situation were some more experienced mappers are merging node information with areas. The rendering is completely different. But we are certainly in no position to give these more experienced mappers any lesson or guidance.
> 
> Look at that area south of Bamako for instance.
> https://www.openstreetmap.org/#map=9/12.3024/-7.6382
> Villages that have names as nodes appear around zoom level 9. But a lot of villages that have been named are no longer labelled until your reach zoom level 13 because their name nodes have been merged into their areas.
> 
> What are we supposed to do with that? Are we supposed to manually undo the work of more experienced mappers and re-add duplicate points to areas that don't have them? That doesn't make any sense since there is no guidance telling mappers NOT to put names into areas.

I used to recommend to switch to another renderer ("Layers" on the right-hand side of the osm.org website), since they used to render area place names, but now it seems they also don't. Does anyone know if their behaviour was changed to be consistent with osm-carto, or am I missing something, or possibly going crazy?

**Comment by jeisenbe** [comment]
Re: " re-add duplicate points to areas that don't have them"

A place node is not a duplicate of an administrative area. For example, consider https://www.openstreetmap.org/node/6474240715 and https://www.openstreetmap.org/relation/112100 - the first is the node which represents the approximate center of the place called "Long Beach" in California, which is located in the southwest corner of the land are of the municipality which is represented by the relation.

<img width="590" alt="Screen Shot 2020-06-18 at 10 43 09" src="https://user-images.githubusercontent.com/42757252/85054178-87ad4100-b150-11ea-921b-1129745041f2.png">

When you want to route to "Long Beach" from Los Angeles, you don't want to end up in the centroid of that municipality polygon: that will be about 5 kilometers northeast of Downtown Long Beach. Both the administrative boundary and the place node are valid representations of different things: one is the legal municipal borders (which exclude the suburb of Signal Hill, for example), while the other is the location of the "centre" of the place. 

Almost any town built near a large body of water will have this problem if there is no node to represent the centre. While the problem is less obvious with small features such as villages, it is still advisable to map these with a node at their commercial or cultural centre. Not only does this make it possible to render the name properly and to provide routing to the correct location, it also represents the named place correctly in the database.

**Comment by jeisenbe** [comment]
Or consider the municipality of New York. Since this is now mapped as an `boundary=administrative` relation with `admin_level=5`, it get's a purple label at ST_PointOnSurface (near the "center" of the polygon), as well as a larger black label near the city hall in Lower Manhattan where the `place=city` node is located:

<img width="769" alt="Screen Shot 2020-06-18 at 10 51 33" src="https://user-images.githubusercontent.com/42757252/85055301-2e461180-b152-11ea-9b4a-d54b53ee40ba.png">

Clearly the node is needed to represent the actual center of the city, which is not located in Cypress Hills Cemeter in suburban Queens, but in downtown Manahattan.

(This is one of the cases where rendering the name of the admin_level=5 feature is not very helpful, but New York City is the only municipality in the USA which is mapped in this way)

**Comment by dawansv** [comment]
@jeisenbe Thank you what you are saying makes a lot of sense. The examples are really good and I would encourage you to add them to the wiki where place names are discussed.

Right now, for Mali, I am getting some help from a more experiece mapper that is reintroducing nodes to the villages so that the rendering will be back to normal. 

Right now we have the place=village tag on both aera and node. I am wondering if that's OK or if we should  only have it on nodes; for the area, the boudary is not administrative (no such thing for villages there), so not sure whether we should use type=boundary or not. I will look in the wiki a bit more and also look for more examples.

**Comment by gravitystorm** [comment]
> since they used to render area place names, but now it seems they also don't. Does anyone know if their behaviour was changed to be consistent with osm-carto, or am I missing something, or possibly going crazy?

The Transport and OpenCycleMap layers both used to render place names from place polygons, but this was dropped a few years ago. The main difficulties were the surprisingly large number of places where the name on the node didn't match the name on the polygon (generally different variations e.g. Saint vs St, but often where place labels were added to admin boundaries which had a different name to the town) but also the much larger number of places where the place categorization didn't match (e.g. place=village on the node, place=city on the polygon). This led to many terrible results, particularly in the USA, where tiny villages were having their names rendered in big fonts on my maps, without any other maps showing that problem.

Unfortunately there's only so long that I could swim against the tide on this topic, so I eventually dropped the polygon place labels and now just use place nodes.

**Comment by nchristensen** [comment]
In my locality, we have named subdivisions with legally defined boundaries (decided at the time of annexation). The land uses in these subdivisions is often mixed so a single `landuse=residential` tag is wrong. At the same time, these areas do not define a separate administrative unit. This leaves `place=neighbourhood` as the best tag for the polygon in my view. I was surprised to find this is not rendered.

**Comment by WebMaster50** [comment]
I came across the same problem and the solution I used was already mentioned in the beginning. Place a way with `place=*` and `name=*` in a relation with a node that has the same place and name tags. You can use this relation technique if the center where the node should be place is not the same as the geometric center of the way. If you specify these two use cases in the wiki for place, it should be fine. 

Take a look at the following examples
1) [Here](https://www.openstreetmap.org/changeset/102752212#map=15/51.2340/6.0440) I used a node to place the node of the city center at a different location.
2) [Here](https://www.openstreetmap.org/changeset/102752212#map=15/51.2340/6.0440), the geometric center is fine for the label so I didn't make a relation, just a way with the propor information. It is a pitty that it does not get rendered but this is correct tagging.
3) [This](https://www.openstreetmap.org/changeset/102752212#map=15/51.2340/6.0440) example clearly shows the implications the choice not to render place names on a way has. I wanted the label to get rendered because it is a well known name. The geometric center is fine but I had to use relations to get it rendered. This should not be needed because it creates unecessary complexity. 

Conclusion, there are already perfect solution available to get around this.

**Comment by twilight1794** [comment]
In many mexican cities neighbourhoods (_colonias_ and _fraccionamientos_) have clear limits inside the city, but now we have to define them as nodes instead of areas. It would be better if place names of areas are rendered.

**Comment by marioxcc** [comment]
Render places mapped as areas. As explained above, there are many cases where the areas are well-defined. Leave it to mappers to determine what those places are.
